tests/test_mgxs_library_condense/results_true.dat

```diff
@@ -1,49 +1,49 @@
-   material  group in nuclide      mean  std. dev.
-0         1         1   total  0.419289    0.01638   material  group in nuclide     mean  std. dev.
-0         1         1   total  0.07774   0.003273   material  group in  group out nuclide      mean  std. dev.
+   group in  material nuclide      mean  std. dev.
```

Comment:
Oh no, that's not fun :-( I've tried to use `OrderedDict` *everywhere* when creating collections of objects to ensure consistent and reproducible orderings of objects since I ran into similar issues in PR #471 (though they weren't specific to Python 3 vs. 2). In particular, the `openmc.MGXS` objects are stored in an `OrderedDict` in `openmc.mgxs.Library`. Similarly, `Materials`, `Cells`, `Universes`, etc. are stored/returned in `OrderedDict` objects in the `Geometry`'s `get_all_materials()`, `get_all_cells()`, `get_all_universes()`, etc. routines. But perhaps I missed a temporary collection of these things somewhere and that is throwing off the tests now.